packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -106,12 +107,18 @@ function NavigationMenuSelector( {
 	const noBlockMenus = ! hasNavigationMenus && hasResolvedNavigationMenus;
 	const menuUnavailable =
 		hasResolvedNavigationMenus && currentMenuId === null;
+	const navMenuHasBeenDeleted = currentMenuId && isNavigationMenuMissing;
```

Comment:
In trunk we have some similar checks and others with `hasResolvedNavigationMenus && isNavigationMenuMissing`. I'm wondering why that difference and mostly why isn't `isNavigationMenuMissing` is doing this check internally in the hook.

@afercia 's comment about `Specifically in this scenario, currentMenuId && isNavigationMenuMissing is true, while menuUnavailable is false.` and why we weren't passing the id here are good ones. Maybe @getdave or @draganescu have more context?

Reply:
> why isn't isNavigationMenuMissing is doing this check internally in the hook

I assume you mean check for hasResolvedNavigationMenus? One of the main problems with the code of the navigation block is that it is a blueprint of all the pitfalls we've ran through as the block evolved. I remember these checks Is* and has* have been part of a "refactoring" to herd some of the repeating code.

> currentMenuId && isNavigationMenuMissing

This check is very valid, I am unsure what the question is, it means: the block has a navigation id assigned but it's missing.

Reply:
The first question is how `currentMenuId && isNavigationMenuMissing` and `hasResolvedNavigationMenus && isNavigationMenuMissing` are different. Related to this is why `isNavigationMenuMissing` not checking this internally, so we wouldn't need the first part of the above checks.

The other two from @afercia are:
1. Why currentMenuId && isNavigationMenuMissing is true, while menuUnavailable is false (this is mentioned in the PR's description.
2. [why we weren't passing the id](https://github.com/WordPress/gutenberg/pull/67009#issuecomment-2476706438)